kotlin/app/src/main/java/com/google/android/gms/samples/wallet/viewmodel/CheckoutViewModel.kt

```diff
@@ -0,0 +1,71 @@
+package com.google.android.gms.samples.wallet.viewmodel
+
+import android.app.Application
+import android.util.Log
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.google.android.gms.common.api.ApiException
+import com.google.android.gms.samples.wallet.util.PaymentsUtil
+import com.google.android.gms.tasks.Task
+import com.google.android.gms.wallet.IsReadyToPayRequest
+import com.google.android.gms.wallet.PaymentData
+import com.google.android.gms.wallet.PaymentDataRequest
+import com.google.android.gms.wallet.PaymentsClient
+
+class CheckoutViewModel(application: Application) : AndroidViewModel(application) {
+
+    /**
+     * A client for interacting with the Google Pay API.
+     *
+     * @see [PaymentsClient](https://developers.google.com/android/reference/com/google/android/gms/wallet/PaymentsClient)
+     */
+    private val paymentsClient: PaymentsClient = PaymentsUtil.createPaymentsClient(application)
+
+    // LiveData with the result of whether the user can pay using Google Pay
+    private val canUseGooglePayData: MutableLiveData<Boolean> by lazy {
+        MutableLiveData<Boolean>().also {
+            canUseGooglePay()
+        }
+    }
+
+    fun getCanUseGooglePay(): LiveData<Boolean> {
+        return canUseGooglePayData
```

Comment:
I think this doesn't have to be a function.

You can just declare a field as
```
val getCanUseGooglePlay: LiveData<Boolean> = _getCanUseGooglePlay
```

Reply:
SG. Updated.
I was following the example in the [view model guide](https://developer.android.com/topic/libraries/architecture/viewmodel#kotlin) in case it was part of any convention:
```
fun getUsers(): LiveData<List<User>> {
        return users
}
```

Reply:
...and if we use public vars instead of functions in Kotlin, why not doing the same thing in Java?

Reply:
I did the same in Java (with `public final` members). I think the added simplicity is worthwhile.